src/Promotion/Action/NthItemUnitDiscount.php

```diff
@@ -59,6 +59,12 @@ public function execute(PromotionSubjectInterface $subject, array $configuration
         /** @var OrderInterface $subject */
         Assert::isInstanceOf($subject, OrderInterface::class);
 
+        try {
+            $this->isConfigurationValid($configuration);
```

Comment:
Instead of throwing exception we should probably just return a false here and just check the condition